repl/src/main/resources/fake_shell.py

```diff
@@ -83,6 +91,79 @@ def execute_reply_internal_error(message, exc_info=None):
     })
 
 
+class JobContextImpl(object):
+    def __init__(self):
+        self.lock = threading.Lock()
+        self.sc = global_dict['sc']
```

Comment:
You might be able to just say `sc` instead of `global_dict['sc']` if I understand the code correctly.